Classes/Private/Manager/Audio/OCTAudioEngine.m

```diff
@@ -127,6 +140,102 @@ - (BOOL)isAudioRunning:(NSError **)error
     return running;
 }
 
+#pragma mark - Buffer Management
+- (void)provideAudioFrames:(const int16_t *)pcm sample_count:(size_t)sample_count channels:(uint8_t)channels sample_rate:(uint32_t)sample_rate
+{
+    // To Do: copy audio frames into buffer.
+}
+
+#pragma mark - Call Backs
+
+- (BOOL)registerInputCallBack:(NSError **)error;
+{
+    AURenderCallbackStruct callbackStruct;
+    callbackStruct.inputProc = inputRenderCallBack;
+    callbackStruct.inputProcRefCon = _ioUnit;
```

Comment:
What is `inputProcRefCon`? Why in this case we are setting it to `ioUnit` and for output to `self`?